packages/permission-controller/src/PermissionController.test.ts

```diff
@@ -2096,6 +2090,7 @@ describe('PermissionController', () => {
 
       controller.updatePermissionsByCaveat(
         CaveatTypes.filterArrayResponse,
+        // @ts-expect-error Abusing types for testing purposes
```

Comment:
This does not seem like an example of destructive testing with invalid inputs. 

We simply need to help the compiler narrow between the `mutator` discriminated union by adding `as const` assertions:

```diff
@@ -2081,16 +2083,15 @@ describe('PermissionController', () => {
       const mutator = () => {
         counter += 1;
         return counter === 1
-          ? { operation: CaveatMutatorOperation.noop }
+          ? { operation: CaveatMutatorOperation.noop as const }
           : {
-              operation: CaveatMutatorOperation.updateValue,
+              operation: CaveatMutatorOperation.updateValue as const,
               value: ['a', 'b'],
             };
       };
 
       controller.updatePermissionsByCaveat(
         CaveatTypes.filterArrayResponse,
-        // @ts-expect-error Abusing types for testing purposes
         mutator,
       );
```